File: 백준/Silver/3474. 교수가 된 현우/교수가 된 현우.cc
#include <bits/stdc++.h>
using namespace std;

int main() {
	cin.tie(NULL); cout.tie(NULL);
	ios_base::sync_with_stdio(false);
	int t, n;

	cin >> t;
	while (t--) {
		int cnt2 = 0, cnt5 = 0;

		cin >> n;
		for (int i = 2; i <= n; i *= 2) {
			cnt2 += n / i;
		}

		for (int i = 5; i <= n; i *= 5) {
			cnt5 += n / i;
		}
		
		cout << min(cnt2, cnt5) << "\n";
	}
	return 0;
}

File: 백준/Silver/3474. 교수가 된 현우/교수가 된 현우.py
import sys

input = sys.stdin.readline

t = int(input())

for _ in range(t):
    n = int(input())
    cnt2 = 0
    cnt5 = 0
    i = 2
    while i <= n:
        cnt2 += n // i
        i *= 2
    i = 5
    while i <= n:
        cnt5 += n // i
        i *= 5
    print(min(cnt2, cnt5))
